### PR TITLE
New features: include, workingDirectory

### PR DESCRIPTION
**Pull request by rcdailey**
* Add gitattributes file

* Implement support for 'include' paths or patterns

* New workingDirectory input to specify cwd for the action

See README for examples / docs.

### CODE REVIEW

**Review by muno92 — CHANGES_REQUESTED** [review]
Thank you for contribution!

These features seem to good, but I think .gitattrubutes config get even better with .editorconfig.
And, next time please separate pull request by one feature.

Thanks.

File: .gitattributes [hunk 1]
```diff
@@ -1 +1,4 @@
-dist/** -diff linguist-generated=true 
\ No newline at end of file
+* text=auto
+*.ts eol=lf
```
[the remark this answers is not in the document]
Comment (rcdailey) [reply .gitattributes:2]: The two are not the same and do not accomplish the same goals.

`.gitattributes` controls, among other things, the line endings that are in your working copy when you check out a file (clone the repo, switch to a new branch, etc). This happens well before you ever run prettier or other tools that use editorconfig. Another important aspect of `.gitattributes` is that it removes the need for `core.crlf` on Windows platforms; which means it guarantees consistent EOL in files checked into the repository itself.

`.editorconfig` will, as it relates here, control how files are saved when open in an editor.

I hope that helps your understanding. It's a complicated topic, so let me know if you have any other questions.

**Comment by rcdailey** [comment]
> Thank you for contribution!
> 
> These features seem to good, but I think .gitattrubutes config get even better with .editorconfig. And, next time please separate pull request by one feature.
> 
> Thanks.

With all due respect, I'm involved in a lot of projects and you're asking me to use more of my free time to maintain an additional 1-2PRs. These changes are arguably small enough to fit into 1 PR without creating too much of a burden on your end. I think FOSS is all about compromise. I thank you for allowing me to submit my changes in 1 PR to allow me the time to offer you this contribution in the first place.

File: .gitattributes (hunk 1 above)
[the remark this answers is not in the document]
Comment (muno92) [reply .gitattributes:2]: Oh, I forgot windows platforms.

**Review by muno92 — APPROVED** [review]

**Pull request merged**